internal/provider/azure/environ.go

```diff
@@ -297,60 +350,35 @@ func (env *azureEnviron) Bootstrap(
 	return result, nil
 }
 
-// initResourceGroup creates a resource group for this environment.
-func (env *azureEnviron) initResourceGroup(ctx envcontext.ProviderCallContext, controllerUUID string, existingResourceGroup, controller bool) error {
+func (env *azureEnviron) resourceTags(controllerUUID string) map[string]string {
 	env.mu.Lock()
+	defer env.mu.Unlock()
 	modelTag := names.NewModelTag(env.config.Config.UUID())
-	resourceTags := tags.ResourceTags(
+	return tags.ResourceTags(
 		modelTag,
 		names.NewControllerTag(controllerUUID),
 		env.config,
 	)
-	env.mu.Unlock()
+}
 
+// createResourceGroup creates a resource group for this environment.
+func (env *azureEnviron) createResourceGroup(ctx context.Context, resourceTags map[string]string) error {
 	resourceGroups, err := env.resourceGroupsClient()
 	if err != nil {
 		return errors.Trace(err)
 	}
-	if existingResourceGroup {
-		logger.Debugf(ctx, "using existing resource group %q for model %q", env.resourceGroup, env.modelName)
-		g, err := resourceGroups.Get(ctx, env.resourceGroup, nil)
-		if err != nil {
-			return env.HandleCredentialError(ctx, errors.Annotatef(err, "checking resource group %q", env.resourceGroup))
-		}
-		if region := toValue(g.Location); region != env.location {
-			return errors.Errorf("cannot use resource group in region %q when operating in region %q", region, env.location)
-		}
-		if err := env.checkResourceGroup(g.ResourceGroup, modelTag); err != nil {
-			return env.HandleCredentialError(ctx, errors.Annotate(err, "validating resource group"))
-		}
-	} else {
-		logger.Debugf(ctx, "creating resource group %q for model %q", env.resourceGroup, env.modelName)
-		if _, err := resourceGroups.CreateOrUpdate(ctx, env.resourceGroup, armresources.ResourceGroup{
-			Location: to.Ptr(env.location),
-			Tags:     toMapPtr(resourceTags),
-		}, nil); err != nil {
-			return env.HandleCredentialError(ctx, errors.Annotate(err, "creating resource group"))
-		}
-	}
-
-	if !controller {
-		// When we create a resource group for a non-controller model,
-		// we must create the common resources up-front. This is so
-		// that parallel deployments do not affect dynamic changes,
-		// e.g. those made by the firewaller. For the controller model,
-		// we fold the creation of these resources into the bootstrap
-		// machine's deployment.
-		if err := env.createCommonResourceDeployment(ctx, resourceTags, nil); err != nil {
-			return errors.Trace(err)
-		}
+	logger.Debugf(ctx, "creating resource group %q for model %q", env.resourceGroup, env.modelName)
+	if _, err := resourceGroups.CreateOrUpdate(ctx, env.resourceGroup, armresources.ResourceGroup{
```

Comment:
Minor suggestion: Would upsert be more concised than CreateOrUpdate?

Reply:
This is actually an Azure SDK API call so we can always ask Microsoft if they want to change it :-)